lib/checks/aria/aria-allowed-role.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Implements allowed roles defined at:
+ * https://www.w3.org/TR/html-aria/#docconformance
+ * https://www.w3.org/TR/SVG2/struct.html#implicit-aria-semantics
+ */
+const { allowImplicit = true, ignoredTags = [] } = options || {};
+
+const tagName = node.tagName.toLowerCase();
+
+// check if the element should be ignored, by an user setting
+if (ignoredTags && ignoredTags.indexOf(tagName) > -1) {
+	return true;
+}
+
+const roleSegments = axe.commons.aria.getRoleSegments(node);
+const implicitRole = axe.commons.aria.implicitRole(node);
+
+/**
+ * Helper object with functions to computed is role allowed for a given tag
+ */
+const elementCheck = {
+	a: (node, role) => {
+		if (node.namespaceURI === 'http://www.w3.org/2000/svg') {
+			return true;
+		}
+		if (node.href) {
+			return axe.commons.aria.isAllowedSubsetRole('aWithHref', role);
+		} else {
```

Comment:
This is an unnecessary block. If we just `return` on the next line, we can keep the happy path closer to the left.